packages/devextreme/js/core/utils/date.js

```diff
@@ -689,6 +689,45 @@ const getMachineTimezoneName = () => {
         : null;
 };
 
+const getRangesByDates = (dates) => {
+    const datesInMilliseconds = dates.map((value) => {
+        const date = new Date(new Date(value).setHours(0, 0, 0, 0)).getTime();
+
+        return date;
+    });
+
+    const sortedDates = datesInMilliseconds.sort((a, b) => a - b);
+
+    const getRange = (date, dates, index) => {
+        const range = date === dates[index - 1]
+            ? [date]
+            : [date, dates[index - 1]];
+
+        return range.map((value) => new Date(value));
+    };
+
+    const msInDay = toMilliseconds('day');
+    const ranges = [];
+
+    let startDate = sortedDates[0];
+
+    for(let i = 1; i <= sortedDates.length; ++i) {
+        const currentDate = sortedDates[i];
+        const previousDate = sortedDates[i - 1];
+        const isNewRange = currentDate - previousDate > msInDay;
+
+        if(isNewRange || i === sortedDates.length) {
+            const range = getRange(startDate, sortedDates, i);
```

Comment:
Don't u think that in this particular case code will be more readable if we get rid of the `getRange` function? Smth like
```javascript
const range = previousDate === startDate
    ? [startDate]
    : [startDate, previousDate];
```
Just an additional idea: maybe it's more clear to change naming `presiousDate -> currentDate`, `currentDate -> nextDate` for making `if` looking like `if (currentDate === startDate)`

Reply:
[10a380daeca82004dfd116401bdefa3c80a5813c]